src/main/java/org/opensearch/performanceanalyzer/collectors/telemetry/RTFHeapMetricsCollector.java

```diff
@@ -0,0 +1,119 @@
+
+    public RTFHeapMetricsCollector() {
+        super(
+                SAMPLING_TIME_INTERVAL,
+                "RTFHeapMetricsCollector",
+                StatMetrics.HEAP_METRICS_COLLECTOR_EXECUTION_TIME,
+                StatExceptionCode.HEAP_METRICS_COLLECTOR_ERROR);
+        this.count = 0;
+    }
+
+    @Override
+    public void collectMetrics(long startTime) {
+        count += 1;
+        metricsRegistry = OpenSearchResources.INSTANCE.getMetricsRegistry();
+        if (metricsRegistry == null) {
+            LOG.error("could not get the instance of MetricsRegistry class");
+            return;
+        }
+
+        initialiseMetrics();
+        GCMetrics.runGCMetrics();
+        LOG.info("Executing collect metrics for RTFHeapMetricsCollector");
+        recordMetrics();
+    }
+
+    private void initialiseMetrics() {
+        GCCollectionEventMetrics =
+                metricsRegistry.createHistogram(
+                        AllMetrics.HeapValue.Constants.COLLECTION_COUNT_VALUE,
+                        "GC Collection Event PA Metrics",
+                        "1");
+
+        GCCollectionTimeMetrics =
+                metricsRegistry.createHistogram(
+                        AllMetrics.HeapValue.Constants.COLLECTION_TIME_VALUE,
+                        "GC Collection Time PA Metrics",
+                        "1");
+
+        HeapUsedMetrics =
+                metricsRegistry.createHistogram(
+                        AllMetrics.HeapValue.Constants.USED_VALUE, "GC Heap Used PA Metrics", "1");
+    }
+
+    private void recordMetrics() {
+        Tags TotYoungGCTag =
+                Tags.create()
+                        .addTag(MemTypeAttributeKey, AllMetrics.GCType.TOT_YOUNG_GC.toString());
+
+        Tags TotFullGCTag =
+                Tags.create().addTag(MemTypeAttributeKey, AllMetrics.GCType.TOT_FULL_GC.toString());
+
+        GCCollectionEventMetrics.record(GCMetrics.getTotYoungGCCollectionCount(), TotYoungGCTag);
+
+        GCCollectionEventMetrics.record(GCMetrics.getTotFullGCCollectionCount(), TotFullGCTag);
+
+        GCCollectionTimeMetrics.record(GCMetrics.getTotYoungGCCollectionTime(), TotYoungGCTag);
+
+        GCCollectionTimeMetrics.record(GCMetrics.getTotFullGCCollectionTime(), TotFullGCTag);
+
+        for (Map.Entry<String, Supplier<MemoryUsage>> entry :
+                HeapMetrics.getMemoryUsageSuppliers().entrySet()) {
+            MemoryUsage memoryUsage = entry.getValue().get();
+            HeapUsedMetrics.record(
+                    memoryUsage.getUsed(),
+                    Tags.create().addTag(MemTypeAttributeKey, entry.getKey()));
+        }
+
+        if (count == 12) {
+            count = 0;
+            for (Map.Entry<String, Supplier<MemoryUsage>> entry :
+                    HeapMetrics.getMemoryUsageSuppliers().entrySet()) {
+                MemoryUsage memoryUsage = entry.getValue().get();
+                metricsRegistry.createGauge(
```

Comment:
Gauge, first initialise once and then it automatically exports the latest value minutely(i.e. our export frequency) so should be ok to write every value to the guage if not aggregation is being done here.

Reply:
I looked into it but gauge does not provide us any method to reuse an already created gauge object.

Reply:
What did you mean by already created gauge object?

Reply:
I might have misunderstood. Are you suggesting to avoid calling metricsRegistry.createGauge() everytime?

Reply:
Both metricsRegistry.createGauge() and counter handling. If you add multiple values to the Gauge metric it will only export the latest value added that's what I guess we are trying to do while emitting the 12th datapoint?

Reply:
Ack. Then I will remove the if block.